packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -154,10 +173,25 @@ export class ProxyContext implements EditableTreeContext {
         return this.getRoot(true);
     }
 
+    public set unwrappedRoot(value: ContextuallyTypedNodeData | undefined) {
+        const rootField = this.getRoot(false);
+        const mapTrees = applyFieldTypesFromContext(this.schema, rootField.fieldSchema, value);
+        const cursors = mapTrees.map(singleMapTreeCursor);
+        if (rootField.length > 0) {
+            rootField.replaceNodes(0, cursors);
+        } else {
+            rootField.insertNodes(0, cursors);
```

Comment:
If I understand correctly, the `insertNodes` case is needed only because of a bug that prevents `replaceNodes` from workin when the field is empty.
Is there a way we could hide that logic inside `replaceNodes` so that callers of it don't have to deal with this?
If not then I think it would be good to add a comment explaining why we're doing this.

Reply:
Fixed.